### PR TITLE
Fix esm2 lora inference

### PR DESCRIPTION
**Pull request by yzhang123**
### Description
fix esm2 lora inference

### Type of changes

- [x]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Refactor
- [x]  Documentation update
- [ ]  Other (please describe):

### CI Pipeline Configuration
Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests
- [INCLUDE_NOTEBOOKS_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_notebooks_tests) - Execute notebook validation tests in pytest
- [INCLUDE_SLOW_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_slow_tests) - Execute tests labelled as slow in pytest for extensive testing

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

* If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
* If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Usage

```python
TODO: Add code snippet
```

### Pre-submit Checklist

 - [ ] I have tested these changes locally
 - [ ] I have updated the documentation accordingly
 - [ ] I have added/updated tests as needed
 - [ ] All existing tests pass successfully

### CODE REVIEW

File: sub-packages/bionemo-esm2/tests/bionemo/esm2/scripts/test_infer_esm2.py
Comment (polinabinder1) [inline sub-packages/bionemo-esm2/tests/bionemo/esm2/scripts/test_infer_esm2.py:1]: Could you add a test case for regression?

**Review by polinabinder1 — CHANGES_REQUESTED** [review]

File: sub-packages/bionemo-esm2/tests/bionemo/esm2/scripts/test_infer_esm2.py
Comment (yzhang123) [reply sub-packages/bionemo-esm2/tests/bionemo/esm2/scripts/test_infer_esm2.py:1]: we don't have lora checkpoint for regression

File: sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/infer_esm2.py [hunk 1]
```diff
@@ -306,9 +348,59 @@ def config_class_type(desc: str) -> Type[BioBertConfig]:
...
+        help="Task type to use for inference.",
+    )
+    parser.add_argument(
+        "--mlp-ft-dropout",
+        type=float,
+        required=False,
+        default=0.25,
+        help="MLP dropout to use for inference.",
+    )
+    parser.add_argument(
```
Comment (trvachov) [inline sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/infer_esm2.py]: High level, why is it that we now have to specify this for inference (including non-LoRA inference?). Was it the case that this should've been part of the interface all along?

Comment (yzhang123) [reply sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/infer_esm2.py]: because for the non lora route that works on a fine-tuned model these configurations are saved as part of the final fine-tuned checkpoint, which are used to correctly initialize the model for inference.

for lora, there is a 2 step process: first we need to initialize the fine-tuned model head with random initialization , then the pretrained checkpoint weights are loaded, the model is transformed into an adapter model, and then the adapter (lora) weights are loaded. these configurations are not stored in the pretrained checkpoint, but there is a way to extract them from the lora checkpoint folder, I will change that.

File: sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/infer_esm2.py [hunk 2]
```diff
@@ -306,9 +348,59 @@ def config_class_type(desc: str) -> Type[BioBertConfig]:
...
+        required=False,
+        default=1,
+        help="MLP target size to use for inference.",
+    )
+    parser.add_argument(
+        "--cnn-dropout",
+        type=float,
+        required=False,
+        default=0.25,
+        help="CNN dropout to use for inference.",
```
Comment (trvachov) [inline sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/infer_esm2.py]: These are just used for LoRA inference, right? If yes, can you just add that to each arg that is needed for LoRA only?

Otherwise the change looks good to me. We can add a train LoRA -> save checkpoint -> load checkpoint for inference test in this PR or future one -- doesn't matter to me.

File: sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/infer_esm2.py (hunk 1 above)
Comment (yzhang123) [reply sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/infer_esm2.py]: i removed the extra arguments

File: sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/infer_esm2.py (hunk 2 above)
Comment (yzhang123) [reply sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/infer_esm2.py]: i removed them

Comment (yzhang123) [reply sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/infer_esm2.py]: i removed the arguments

**Review by jstjohn — APPROVED** [review]
🚢
